src/org/opendatakit/briefcase/ui/reused/AggregateServerConnectionDialog.java

```diff
@@ -0,0 +1,68 @@
+package org.opendatakit.briefcase.ui.reused;
+
+import java.awt.Cursor;
+
+import org.opendatakit.briefcase.model.ServerConnectionInfo;
+import org.opendatakit.briefcase.model.TerminationFuture;
+import org.opendatakit.briefcase.ui.MessageStrings;
+import org.opendatakit.briefcase.ui.ODKOptionPane;
+import org.opendatakit.briefcase.util.ServerConnectionTest;
+
+public class AggregateServerConnectionDialog {
+  final AggregateServerConnectionDialogForm form;
+  private boolean asTarget;
+  private boolean isSuccessful;
+  private ServerConnectionInfo serverInfo = null;
+  private TerminationFuture terminationFuture = new TerminationFuture();
+
+  AggregateServerConnectionDialog(AggregateServerConnectionDialogForm aggregateServerConnectionDialogForm, boolean asTarget) {
+    form = aggregateServerConnectionDialogForm;
+    this.asTarget = asTarget;
+
+    form.addConnectButtonCallback(e -> verifyParams());
```

Comment:
The convention we're following is `onConnect( <some callback here> )`

Reply:
Also, the callback should receive the user's input from the form in some structured object. Therefor, the callback should be of type `Consumer<ServerConnectionParams>` or something like that.